troncos/traces/decorate.py

```diff
@@ -99,40 +107,46 @@ def myfunc1()
     def myfunc2()
         return "This will be traced using a custom provider"
     """
-
     if len(args) > 1:
         raise RuntimeError("Invalid usage of decorator")
     if len(args) == 1 and (callable(args[0]) or asyncio.iscoroutinefunction(args[0])):
         func = args[0]
         assert func
-        return _trace_function(func, name, attributes, tracer_provider)
+        return _trace_function(func, name, resource, attributes, tracer_provider)
     else:
         # No args
         def _inner(f: Callable[P, R]) -> Callable[P, R]:
-            return _trace_function(f, name, attributes, tracer_provider)
+            return _trace_function(f, name, resource, attributes, tracer_provider)
 
         return _inner
 
 
 def trace_block(
     name: str,
+    resource: str | None = None,
+    *,
     attributes: Attributes | None = None,
     tracer_provider: opentelemetry.trace.TracerProvider | None = None,
 ) -> _GeneratorContextManager[opentelemetry.trace.Span]:
     """
     Trace using a with statement. You can supply a tracer provider, if none is supplied,
     the global tracer provider will be used. Example:
 
-    with trace_block(name="my block", attributes={"some": "attribute"}):
+    with trace_block("cool.block", "data!", attributes={"some": "attribute"}):
```

Comment:
Now this makes total sense. You have an operation name that is acting on some resource :+1: 

How does this make sense on the `trace_function` decorator? Is `data!`, aka `resource` dynamic? 

Can you add a similar example to the decorator docs?